### PR TITLE
Doc update

### LINKED ISSUE #17: Data field comments improvements
The field comments for Registry, Project, etc. shall describe in text the OpenAPIV3 validation rules so they will be visible in the generated swagger doc too.

### LINKED ISSUE #18: Godoc comment improvements
The go comment docs shall be fixed throughout the project so that it looks good in godoc.

### PR DESCRIPTION
**Pull request by szamuboy**
**H U G E** PR with a lot of changes:
- code documentation improvements
- removal of unneeded code
- simplifying complex things
- refactoring some parts
- following naming conventions

### CODE REVIEW

File: pkg/config/apiobjectstore.go [hunk 1]
```diff
@@ -53,17 +53,45 @@ func init() {
...
+// Registries and Scanners).
 type ApiObjectStore struct {
 	store      map[schema.GroupVersionKind][]runtime.Object
 	serializer *json.Serializer
+	path       string
+}
+
+// WriteManifest serializes the object specified by the obj parameter. The
+// filename parameter specifies the name of the file to be created. The path
+// where the file is created is determined the ApiObjectStore.
```
Comment (Kukkerem) [inline pkg/config/apiobjectstore.go]: I don't really understand this sentence. It means that the directory path is determined when we created the ApiObjectStore object, or we can determine the ApiObjectStore object with the path? (Or both of them, which means that it's OK :) )

File: pkg/config/apiobjectstore.go [hunk 2]
```diff
@@ -53,17 +53,45 @@ func init() {
...
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
-func (aos *ApiObjectStore) GetSerializer() *json.Serializer {
-	return aos.serializer
+// RemoveManifest removes the file from the filesystem. The path where the file
+// is removed from is determined the ApiObjectStore.
```
Comment (Kukkerem) [inline pkg/config/apiobjectstore.go]: Same as above.

File: pkg/globalregistry/reconciler/memberstatus.go [hunk 3]
```diff
@@ -103,16 +109,20 @@ type persistMemberCredentials struct {
...
+	RemoveManifest(filename string) error
+}
+
 func (pmc *persistMemberCredentials) Perform(ctx context.Context) error {
-	path := ctx.Value(SideEffectPath)
-	if path == nil {
-		return fmt.Errorf("context shall contain SideEffectPath")
+	sideEffectManipulatorCtx := ctx.Value(SideEffectManifestManipulator)
+	if sideEffectManipulatorCtx == nil {
+		return fmt.Errorf("contaxt shall contain SideEffectManifestManipulator")
```
Comment (Kukkerem) [inline pkg/globalregistry/reconciler/memberstatus.go]: contaxt typo

File: pkg/globalregistry/reconciler/memberstatus.go [hunk 4]
```diff
@@ -190,17 +190,21 @@ type removeMemberCredentials struct {
 var _ SideEffect = &removeMemberCredentials{}
 
 func (rmc *removeMemberCredentials) Perform(ctx context.Context) error {
-	path := ctx.Value(SideEffectPath)
-	if path == nil {
-		return fmt.Errorf("context shall contain SideEffectPath")
+	sideEffectManipulatorCtx := ctx.Value(SideEffectManifestManipulator)
+	if sideEffectManipulatorCtx == nil {
+		return fmt.Errorf("contaxt shall contain SideEffectManifestManipulator")
```
Comment (Kukkerem) [inline pkg/globalregistry/reconciler/memberstatus.go]: contaxt

File: pkg/config/apiobjectstore.go (hunk 1 above)
Comment (szamuboy) [reply pkg/config/apiobjectstore.go]: comment doc updated

File: pkg/config/apiobjectstore.go (hunk 2 above)
Comment (szamuboy) [reply pkg/config/apiobjectstore.go]: comment doc updated

File: pkg/globalregistry/reconciler/memberstatus.go (hunk 3 above)
Comment (szamuboy) [reply pkg/globalregistry/reconciler/memberstatus.go]: fixed

File: pkg/globalregistry/reconciler/memberstatus.go (hunk 4 above)
Comment (szamuboy) [reply pkg/globalregistry/reconciler/memberstatus.go]: fixed

File: pkg/globalregistry/reconciler/replicationrulestatus.go [hunk 5]
```diff
@@ -22,10 +23,18 @@ import (
...
-	Direction          globalregistry.ReplicationDirection `json:"direction"`
+
+	// RemoteRegistryName indicates the name of the remote registry which
+	// the current registry shall synchronize with.
+	RemoteRegistryName string `json:"name"`
+
+	// Trigger describes the event that shall trigger the replication.
+	Trigger globalregistry.ReplicationTrigger `json:"trigger"`
+
+	// Direction shows whether the replication is of type pull or pull.
```
Comment (Kukkerem) [inline pkg/globalregistry/reconciler/replicationrulestatus.go:36]: pull or push?

Comment (Kukkerem) [inline pkg/globalregistry/reconciler/replicationrulestatus.go]: pull or push?

File: pkg/globalregistry/reconciler/scannerstatus.go [hunk 6]
```diff
@@ -6,19 +22,26 @@ import (
 	"github.com/kubermatic-labs/registryman/pkg/globalregistry"
 )
 
+// ScannerStatus specifies the status of a project's external vulnerability scanner.
 type ScannerStatus struct {
+
+	// Name of the scanner
```
Comment (Kukkerem) [inline pkg/globalregistry/reconciler/scannerstatus.go]: .

File: pkg/globalregistry/project.go [hunk 7]
```diff
@@ -41,25 +47,57 @@ type ProjectMemberCredentials struct {
...
+// the management of the projects.
 type ProjectAPI interface {
-	Create(name string) (Project, error)
-	GetByName(name string) (Project, error)
+
+	// List returns the list of the projects managed by the registry.
 	List() ([]Project, error)
+
+	// GetByName returns the project with the given name. If no project is
+	// present with the given name nil, nil is returned.
```
Comment (Kukkerem) [inline pkg/globalregistry/project.go]: name ~~nil~~

File: pkg/globalregistry/project.go [hunk 8]
```diff
@@ -41,25 +47,57 @@ type ProjectMemberCredentials struct {
...
+	GetByName(name string) (Project, error)
+
+	// Create create a new project with the given name.
```
Comment (Kukkerem) [inline pkg/globalregistry/project.go]: creates

File: pkg/globalregistry/replication.go [hunk 9]
```diff
@@ -83,15 +90,33 @@ func (rt ReplicationTrigger) MarshalText() ([]byte, error) {
 	return []byte(s), err
 }
 
+// ReplicationRule interface declares the methods that can be used to manipulate
+// the replication rule of a project.
 type ReplicationRule interface {
+	// GetProjectName returns the name of the project that is subject to the
```
Comment (Kukkerem) [inline pkg/globalregistry/replication.go:97]: new line before comment

File: pkg/globalregistry/reconciler/memberstatus_test.go [hunk 10]
```diff
@@ -5,14 +5,15 @@
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at
 
-http://www.apache.org/licenses/LICENSE-2.0
+  http://www.apache.org/licenses/LICENSE-2.0
```
Comment (Kukkerem) [inline pkg/globalregistry/reconciler/memberstatus_test.go]: +1 space

**Review by Kukkerem — CHANGES_REQUESTED** [review]
missed files:
- globalregistry/storage.go
- hack/tools.go
- quay/*

**Comment by szamuboy** [comment]
> missed files:
> 
>     * globalregistry/storage.go
> 
>     * hack/tools.go
> 
>     * quay/*

quay going to be removed, it is not really part of the project

**Review by Kukkerem — APPROVED** [review]

**Review by cataclyst23 — APPROVED** [review]

**Pull request closed without merging**